core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/CircuitBreakerException.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.user.core;
+
+/**
+ * The exception to throw when the Circuit Breaker triggers for user stores.
+ */
+public class CircuitBreakerException extends UserStoreException {
```

Comment:
Exception name does not indicate that the circuit is broken. I'm not sure if this is the correct term, but something like OpenCircuitBreakerException would be more meaningful.